libcnb-data/src/buildpackage.rs

```diff
@@ -0,0 +1,257 @@
+use crate::buildpackage::PlatformOs::Linux;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::path::PathBuf;
+use uriparse::{URIReference, URIReferenceError};
+
+fn platform_default() -> Platform {
+    Platform::default()
+}
+
+/// Data structure for the Buildpackage configuration schema (package.toml) for a buildpack.
+///
+/// Representation of [package.toml](https://buildpacks.io/docs/reference/config/package-config/).
+///
+/// # Example
+/// ```
+/// use libcnb_data::buildpackage::Buildpackage;
+///
+/// let toml_str = r#"
+/// [buildpack]
+/// uri = "."
+///
+/// [[dependencies]]
+/// uri = "libcnb:buildpack_id"
+///
+/// [[dependencies]]
+/// uri = "../relative/path"
+///
+/// [[dependencies]]
+/// uri = "/absolute/path"
+///
+/// [[dependencies]]
+/// uri = "docker://docker.io/heroku/procfile-cnb:2.0.0"
+///
+/// [platform]
+/// os = "windows"
+/// "#;
+///
+/// let buildpackage = toml::from_str::<Buildpackage>(toml_str).unwrap();
+/// ```
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Buildpackage {
+    pub buildpack: BuildpackageBuildpack,
+    #[serde(default)]
+    pub dependencies: Vec<BuildpackageDependency>,
+    #[serde(default = "platform_default")]
+    pub platform: Platform,
+}
+
+impl Default for Buildpackage {
+    fn default() -> Self {
+        Buildpackage {
+            buildpack: BuildpackageBuildpack::try_from(".").expect("This must be a valid uri"),
+            dependencies: vec![],
+            platform: Platform::default(),
+        }
+    }
+}
+
+/// The buildpack to package. If the `uri` field is a relative path it will be relative to the `package.toml` file.
```

Comment:
Should the "If the `uri` field is a relative path it will be relative to the `package.toml` file." bit rather be a doc comment on the `uri` field directly?